main/src/ca/uwaterloo/flix/language/ast/SyntaxTree.scala

```diff
@@ -0,0 +1,504 @@
+
+      case object LetImport extends Expr
+
+      case object LetMatch extends Expr
+
+      case object LetRecDef extends Expr
+
+      case object Literal extends Expr
+
+      case object LiteralArray extends Expr
+
+      case object LiteralList extends Expr
+
+      case object LiteralMap extends Expr
+
+      case object LiteralRecord extends Expr
+
+      case object LiteralRecordField extends Expr
+
+      case object LiteralSet extends Expr
+
+      case object LiteralVector extends Expr
+
+      case object Match extends Expr
+
+      case object MatchRule extends Expr
+
+      case object NewObject extends Expr
+
+      case object Open extends Expr
+
+      case object OpenAs extends Expr
+
+      case object Paren extends Expr
+
+      case object ParYield extends Expr
+
+      case object ParYieldFragment extends Expr
+
+      case object RecordOperation extends Expr
+
+      case object RecordOpExtend extends Expr
+
+      case object RecordOpRestrict extends Expr
+
+      case object RecordOpUpdate extends Expr
+
+      case object RecordSelect extends Expr
+
+      case object Ref extends Expr
+
+      case object Scope extends Expr
+
+      case object ScopeName extends Expr
+
+      case object Spawn extends Expr
+
+      case object Statement extends Expr
+
+      case object Static extends Expr
+
+      case object StringInterpolation extends Expr
+
+      case object Try extends Expr
+
+      case object TryCatchBody extends Expr
+
+      case object TryCatchRule extends Expr
+
+      case object TryWithBody extends Expr
+
+      case object TryWithRule extends Expr
+
+      case object Tuple extends Expr
+
+      case object TypeMatch extends Expr
+
+      case object TypeMatchRule extends Expr
+
+      case object Unary extends Expr
+
+      case object UncheckedCast extends Expr
+
+      case object UncheckedMaskingCast extends Expr
+
+      case object Use extends Expr
+
+      case object Without extends Expr
+
+    }
+
+    /////////////// TYPES       ///////////////
+    sealed trait Type extends TreeKind {
+      override def debug_name: Option[String] = Some("Type")
+    }
+
+    object Type {
+      /**
+       * A marker kind used to wrap nested types.
+       * For instance on a tuple type "(Int32, Bool)" you would do
+       * Type
+       *   Tuple
+       *     Type
+       *       Ident
+       *     Type
+       *       Ident
+       */
+      case object Type extends Type
+
+      case object Apply extends Type
+
+      case object Argument extends Type
+
+      case object Arguments extends Type
+
+      case object Ascribe extends Type
+
+      case object Binary extends Type
+
+      case object CaseSet extends Type
+
+      case object Constant extends Type
+
+      case object Constraint extends Type
+
+      case object Constraints extends Type
+
+      case object Derivations extends Type
+
+      case object EffectSet extends Type
+
+      case object Function extends Type
```

Comment:
Usually called Arrow, but perhaps this is one case where we should keep your name 👍

Reply:
I think `Function` makes sense here. It better communicates what the construct is in my opinion.

Reply:
Yeah, its just that in 500 other places we call it arrow.

Reply:
I agree it should have been called Function, but for consistency, let us rename it to Arrow. Thanks!